### ISSUE TITLE
Review unit tests for speed

### ISSUE THREAD

**Issue by danielballan** [bluesky/tiled#325]
The unit tests have gotten slower, locally, and _much_ slower on GitHub. It seems that we may be surpassing some threshold where GH Actions begins throttling us and making are slow tests even slower.

Let's look at speeding up the tests by:

* Using server instances shared across a whole module where the downside of coupled tests is low
* Reducing example data size

**Comment by danielballan** [comment]
I learned that pytest provides [a CLI option](https://docs.pytest.org/en/latest/how-to/usage.html#profiling-test-execution-duration) for profiling test speed. This command shows all tests that take longer than one second to run, sorted from longest to shortest:

```
pytest --durations=0 --durations-min=1.0
```

These are the results on my local machine. Presumably results on CI are even slower, especially for the tests that involve I/O from disk.

```
98.32s call     tiled/_tests/test_xarray.py::test_wide_table_optimization_off
45.53s call     tiled/_tests/test_xarray.py::test_url_limit_handling
34.73s call     tiled/_tests/test_writing.py::test_write_large_array_full
27.71s call     tiled/_tests/test_xarray.py::test_dataset_column_access[image]
19.58s call     tiled/_tests/test_tiff.py::test_tiff_sequence[slice_input2-correct_shape2]
17.71s call     tiled/_tests/test_authentication.py::test_session_limit
15.35s call     tiled/_tests/test_sort.py::test_sort[letter-sorted_list0]
13.86s call     tiled/_tests/test_authentication.py::test_api_keys
13.46s call     tiled/_tests/test_object_cache.py::test_detect_content_changed_or_removed
12.86s call     tiled/_tests/test_queries.py::test_structure_families
11.51s call     tiled/_tests/test_hdf5.py::test_from_file
11.46s call     tiled/_tests/test_tiff.py::test_tiff_sequence_with_directory_walker
10.86s call     tiled/_tests/test_xarray.py::test_xarray_dataset[ragged]
10.44s call     tiled/_tests/test_writing.py::test_write_array_chunked
10.24s call     tiled/_tests/test_xarray.py::test_dataset_column_access[weather]
10.05s call     tiled/_tests/test_export.py::test_export_table[spreadsheet.xlsx-numpy]
9.38s call     tiled/_tests/test_tiff.py::test_tiff_sequence[slice_input4-correct_shape4]
8.99s call     tiled/_tests/test_xarray.py::test_xarray_dataset[image]
8.83s call     tiled/_tests/test_indexers.py::test_indexers
8.49s call     tiled/_tests/test_xarray.py::test_xarray_dataset[weather]
8.22s call     tiled/_tests/test_tiff.py::test_tiff_sequence[None-correct_shape0]
8.07s call     tiled/_tests/test_xarray.py::test_wide_table_optimization
7.85s call     tiled/_tests/test_authentication.py::test_refresh_flow
7.71s call     tiled/_tests/test_directory_walker.py::test_mimetype_detection_hook
7.39s call     tiled/_tests/test_writing.py::test_write_sparse_chunked
7.25s call     tiled/_tests/test_client_cache.py::test_when_full
7.23s call     tiled/_tests/test_directory_walker.py::test_files_config_alias
6.92s call     tiled/_tests/test_writing.py::test_write_dataframe_partitioned
6.72s call     tiled/_tests/test_writing.py::test_write_sparse_full
6.67s call     tiled/_tests/test_config.py::test_root
6.57s call     tiled/_tests/test_xarray.py::test_xarray_dataset[wide]
6.44s call     tiled/_tests/test_client_cache.py::test_too_large
6.43s call     tiled/_tests/test_tiff.py::test_tiff_sequence[slice_input3-correct_shape3]
6.43s call     tiled/_tests/test_sparse.py::test_sparse_single_chunk
6.42s call     tiled/_tests/test_writing.py::test_write_dataframe_full
6.15s call     tiled/_tests/test_tiff.py::test_tiff_sequence_block[block_input0-correct_shape0]
5.97s call     tiled/_tests/test_client_cache.py::test_must_revalidate[on_disk]
5.88s call     tiled/_tests/test_size_limit.py::test_array
5.85s call     tiled/_tests/test_writing.py::test_write_array_full
5.84s call     tiled/_tests/test_validation.py::test_validators
5.77s call     tiled/_tests/test_directory_walker.py::test_subdirectory_handler
5.74s call     tiled/_tests/test_export.py::test_serialization_error_hdf5_metadata
5.65s call     tiled/_tests/test_client_cache.py::test_content_stale_at[on_disk-dask]
5.60s call     tiled/_tests/test_tiff.py::test_rgb
5.59s call     tiled/_tests/test_structured_array.py::test_read
5.58s call     tiled/_tests/test_sparse.py::test_sparse_multi_chunk
5.31s call     tiled/_tests/test_client_cache.py::test_offline[in_memory-dask]
5.15s call     tiled/_tests/test_client_cache.py::test_entries_stale_at[on_disk-numpy]
4.91s call     tiled/_tests/test_tiff.py::test_tiff_sequence[0-correct_shape1]
4.65s call     tiled/_tests/test_hdf5.py::test_from_multiple
4.56s call     tiled/_tests/test_sort.py::test_sort[number-sorted_list1]
4.52s call     tiled/_tests/test_specs.py::test_spec_is_converted_to_str
4.46s call     tiled/_tests/test_specs.py::test_specs
4.44s call     tiled/_tests/test_size_limit.py::test_dataframe
4.42s call     tiled/_tests/test_search.py::test_compound_key_into_results
4.39s call     tiled/_tests/test_sort.py::test_sort_missing
4.23s call     tiled/_tests/test_inlined_contents.py::test_too_wide_for_inline
4.20s call     tiled/_tests/test_object_cache.py::test_object_cache_hit_and_miss
4.19s call     tiled/_tests/test_sort.py::test_sort_two_columns
4.19s call     tiled/_tests/test_search.py::test_compound_search
4.11s call     tiled/_tests/test_sort.py::test_sort_sparse
4.10s call     tiled/_tests/test_search.py::test_search[cat-expected_keys4]
4.02s call     tiled/_tests/test_authentication.py::test_multiple_providers
4.01s call     tiled/_tests/test_hdf5.py::test_from_group
4.00s call     tiled/_tests/test_search.py::test_search[orange-expected_keys2]
3.97s call     tiled/_tests/test_queries.py::test_regex
3.97s call     tiled/_tests/test_search.py::test_key_into_results
3.91s call     tiled/_tests/test_object_cache.py::test_cache_size_absolute
3.91s call     tiled/_tests/test_authentication.py::test_key_rotation
3.89s call     tiled/_tests/test_dataframe.py::test_dataframe_column_access
3.88s call     tiled/_tests/test_queries.py::test_specs
3.82s call     tiled/_tests/test_client_cache.py::test_download[in_memory-numpy]
3.79s call     tiled/_tests/test_object_cache.py::test_object_cache_disabled
3.78s call     tiled/_tests/test_search.py::test_search[yellow-expected_keys1]
3.77s call     tiled/_tests/test_search.py::test_search[dog-expected_keys3]
3.74s call     tiled/_tests/test_indexers.py::test_deprecated_indexer_accessors
3.72s call     tiled/_tests/test_authentication.py::test_password_auth
3.67s call     tiled/_tests/test_queries.py::test_noteq
3.66s call     tiled/_tests/test_search.py::test_search[red-expected_keys0]
3.65s call     tiled/_tests/test_queries.py::test_in
3.61s call     tiled/_tests/test_queries.py::test_full_text
3.58s call     tiled/_tests/test_export.py::test_path_as_Path_or_string
3.57s call     tiled/_tests/test_queries.py::test_comparison
3.52s call     tiled/_tests/test_queries.py::test_eq
3.50s call     tiled/_tests/test_authentication.py::test_api_key_limit
3.50s call     tiled/_tests/test_queries.py::test_notin
3.42s call     tiled/_tests/test_export.py::test_export_weather_data_var[numbers.csv-dask]
3.40s call     tiled/_tests/test_history.py::test_history
3.36s call     tiled/_tests/test_hdf5.py::test_from_file_with_vlen_str_dataset
3.35s call     tiled/_tests/test_queries.py::test_contains
3.34s call     tiled/_tests/test_export.py::test_formats
3.33s call     tiled/_tests/test_object_cache.py::test_cache_size_relative
3.28s call     tiled/_tests/test_authentication.py::test_admin
3.19s call     tiled/_tests/test_export.py::test_export_table[spreadsheet.xlsx-dask]
3.15s call     tiled/_tests/test_client_cache.py::test_offline[on_disk-dask]
3.11s call     tiled/_tests/test_export.py::test_export_table[numbers.csv-dask]
3.09s call     tiled/_tests/test_export.py::test_export_2d_array[image.tiff-dask]
3.07s call     tiled/_tests/test_export.py::test_export_weather_data_var[numbers.csv-numpy]
3.06s call     tiled/_tests/test_export.py::test_export_weather_all[test.h5-numpy]
3.00s call     tiled/_tests/test_client_cache.py::test_offline[in_memory-numpy]
2.97s call     tiled/_tests/test_export.py::test_export_2d_array[image.png-numpy]
2.97s call     tiled/_tests/test_client_cache.py::test_offline[on_disk-numpy]
2.96s call     tiled/_tests/test_export.py::test_export_2d_array[numbers.csv-dask]
2.93s call     tiled/_tests/test_config.py::test_single_deeply_nested
2.93s call     tiled/_tests/test_export.py::test_export_2d_array[image.tiff-numpy]
2.92s call     tiled/_tests/test_export.py::test_export_2d_array[numbers.csv-numpy]
2.92s call     tiled/_tests/test_export.py::test_export_table[numbers.csv-numpy]
2.89s call     tiled/_tests/test_client_cache.py::test_metadata_stale_at[in_memory-numpy]
2.88s call     tiled/_tests/test_export.py::test_export_2d_array[image.png-dask]
2.87s call     tiled/_tests/test_directory_walker.py::test_same_filename_separate_directory
2.82s call     tiled/_tests/test_export.py::test_export_weather_all[test.h5-dask]
2.78s call     tiled/_tests/test_directory_walker.py::test_sort
2.77s call     tiled/_tests/test_directory_walker.py::test_collision_after_startup
2.77s call     tiled/_tests/test_directory_walker.py::test_remove_and_re_add
2.75s call     tiled/_tests/test_config.py::test_many_nested
2.71s call     tiled/_tests/test_distinct.py::test_distinct
2.68s call     tiled/_tests/test_directory_walker.py::test_item_added
2.66s call     tiled/_tests/test_client_cache.py::test_metadata_stale_at[on_disk-numpy]
2.60s call     tiled/_tests/test_dataframe.py::test_dataframe_single_partition
2.56s call     tiled/_tests/test_client.py::test_configurable_timeout
2.54s call     tiled/_tests/test_client_cache.py::test_content_stale_at[in_memory-numpy]
2.52s call     tiled/_tests/test_directory_walker.py::test_from_directory
2.39s call     tiled/_tests/test_directory_walker.py::test_item_removed
2.39s call     tiled/_tests/test_client_cache.py::test_content_stale_at[on_disk-numpy]
2.38s call     tiled/_tests/test_dataframe.py::test_dataframe_basic
2.34s call     tiled/_tests/test_directory_walker.py::test_collision_at_startup
2.33s call     tiled/_tests/test_client_cache.py::test_metadata_stale_at[on_disk-dask]
2.27s call     tiled/_tests/test_custom_format.py::test_xdi_round_trip
2.24s call     tiled/_tests/test_client_cache.py::test_entries_stale_at[on_disk-dask]
2.23s call     tiled/_tests/test_config.py::test_single_nested
2.23s call     tiled/_tests/test_client_cache.py::test_metadata_stale_at[in_memory-dask]
2.11s call     tiled/_tests/test_client_cache.py::test_download[on_disk-numpy]
2.09s call     tiled/_tests/test_client_cache.py::test_content_stale_at[in_memory-dask]
2.08s call     tiled/_tests/test_client_cache.py::test_download[in_memory-dask]
2.05s call     tiled/_tests/test_client_cache.py::test_entries_stale_at[in_memory-dask]
1.98s call     tiled/_tests/test_client_cache.py::test_entries_stale_at[in_memory-numpy]
1.88s call     tiled/_tests/test_client_cache.py::test_must_revalidate[in_memory]
1.75s call     tiled/_tests/test_client_cache.py::test_download[on_disk-dask]
1.75s call     tiled/_tests/test_client_cache.py::test_download_with_no_cache
1.53s call     tiled/_tests/test_array.py::test_array_interface
1.52s call     tiled/_tests/test_access_control.py::test_top_level_access_control
1.31s call     tiled/_tests/test_routes.py::test_meta_routes[/]
1.24s call     tiled/_tests/test_routes.py::test_meta_routes[/docs]
1.14s call     tiled/_tests/test_authentication.py::test_revoke_session
1.02s call     tiled/_tests/test_array.py::test_scalar_dtypes[c]

```

**Comment by danielballan** [comment]
Hm, the reported numbers don't seem to match the actual runtime.

**Comment by danielballan** [comment]
Tests are running faster again. Closed by #333.
